packages/roosterjs-editor-plugins/lib/Paste/Paste.ts

```diff
@@ -131,7 +131,18 @@ export default class Paste implements EditorPlugin {
 
             for (let node of nodes) {
                 if (mergeCurrentFormat) {
-                    this.applyTextFormat(node, clipboardData.originalFormat);
+                    this.applyToElements(node, this.applyFormatting(clipboardData.originalFormat));
+                }
+                if (this.editor.isDarkMode()) {
+                    // either use their paste handler or ours, but check it here.
+                    this.applyToElements(node, (element) => {
+                        if (this.editor.getDarkModeOptions().onExternalContentTransform) {
+                            this.editor.getDarkModeOptions().onExternalContentTransform(element);
+                        } else {
+                            element.style.color = null;

```

Comment:
Split this into its own file? Or premature?

Reply:
I think it'd be more elegant if the options argument had a this as a function by default rather than having to check at the callsite...

but we don't have strict null checks turned on so arguably that's less safe

Reply:
That would be very nice, but again, since no strict null checking we're kinda hosed for that. Leaving as is unless you feel like there's a better response, and we can split it out in the future.